### PR TITLE
PODS-1756 : Part II establisher transformation added 

### PR DESCRIPTION
**Pull request by abhijeetardale**
PODS-1756 : Part II establisher transformation added

### CODE REVIEW

File: app/models/details/transformation/CompanyDetailsRows.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+package models.details.transformation
+
+import javax.inject.Inject
+import models.details._
+import utils.CountryOptions
+import viewmodels.{AnswerRow, AnswerSection, SuperSection}
+
+import scala.language.implicitConversions
+
+case class CompanyDetailsRows[I <: CompanyDetails] @Inject()(countryOptions: CountryOptions) extends TransformedElement[I] {
```
Comment (grantrobson) [inline app/models/details/transformation/CompanyDetailsRows.scala]: This doesn't need to be a case class as there is no state.

File: app/models/details/transformation/CompanyDetailsRows.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  override val entityType: String = "company"
```
Comment (grantrobson) [inline app/models/details/transformation/CompanyDetailsRows.scala]: Make vals and defs which aren't used outside this class protected.

File: app/models/details/transformation/EstablisherInfoSection.scala [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+package models.details.transformation
+
+import javax.inject.Inject
+import models.details._
+import viewmodels.MasterSection
+
+import scala.language.implicitConversions
+
+case class EstablisherInfoSection @Inject()(individualInfoRows :IndividualInfoRows[IndividualInfo],
```
Comment (grantrobson) [inline app/models/details/transformation/EstablisherInfoSection.scala]: This doesn't need to be a case class as it has not internal state.

File: app/models/details/transformation/IndividualInfoRows.scala [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+import javax.inject.Inject
+import models.details._
+import org.joda.time.LocalDate
+import utils.{CountryOptions, DateHelper}
+import viewmodels.{AnswerRow, AnswerSection, SuperSection}
+
+import scala.language.implicitConversions
+
+case class IndividualInfoRows[I <: IndividualInfo] @Inject()(countryOptions: CountryOptions) extends TransformedElement[I] {
```
Comment (grantrobson) [inline app/models/details/transformation/IndividualInfoRows.scala]: This doesn't need to be a case class as it has not internal state. Also, make protected or private anything not used outside the class.

File: app/models/details/transformation/IndividualInfoRows.scala [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  def transformSuperSection(data: I, heading : Option[String]): SuperSection = {
+
+    val superHeading = if(heading.isEmpty){Some(fullName(data.personalDetails.name))} else heading
+
+    val answerHeading = if(heading.isEmpty){None} else {Some(fullName(data.personalDetails.name))}
+
+    SuperSection(superHeading, Seq(AnswerSection(answerHeading, transformRows(data))))
+  }
+
+  override def transformSuperSection(data: I): SuperSection = {
```
Comment (grantrobson) [inline app/models/details/transformation/IndividualInfoRows.scala]: Remove unnecessary brackets from methods.

File: app/models/details/transformation/IndividualInfoRows.scala [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      contactRows(data.contact)
+  }
+
+  private def dateOfBirthRows(personalDetails: PersonalInfo): Seq[AnswerRow]  = {
+
+    Seq(transformRow(label = "messages__psaSchemeDetails__individual_date_of_birth", answer = Seq(
+      DateHelper.formatDate(new LocalDate(personalDetails.dateOfBirth)))))
+  }
+
+  private def ninoRows(ninoStr: Option[String]): Seq[AnswerRow]  = {
```
Comment (grantrobson) [inline app/models/details/transformation/IndividualInfoRows.scala]: You could nest this method inside transform rows as it's only used there and will make things clearer. Also dateofbirthrows method.

File: app/models/details/transformation/PartnershipDetailsRows.scala [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+package models.details.transformation
+
+import javax.inject.Inject
+import models.details._
+import utils.CountryOptions
+import viewmodels.{AnswerRow, AnswerSection, SuperSection}
+
+import scala.language.implicitConversions
+
+case class PartnershipDetailsRows[I <: PartnershipDetails] @Inject()(countryOptions: CountryOptions) extends TransformedElement[I] {
```
Comment (grantrobson) [inline app/models/details/transformation/PartnershipDetailsRows.scala]: This doesn't need to be a case class.

File: app/models/details/transformation/SchemeDetailsSection.scala [hunk 8]
```diff
@@ -17,16 +17,18 @@
...
 import utils.CountryOptions
-import viewmodels.AnswerRow
+import viewmodels.{AnswerRow, AnswerSection, SuperSection}
 
 import scala.language.implicitConversions
 
-//noinspection SpellCheckingInspection
-//scalastyle:off method.length
-case class SchemeDetailsRows[I <: SchemeDetails] @Inject()(countryOptions: CountryOptions) extends TransformedElement[I] {
+case class SchemeDetailsSection[I <: SchemeDetails] @Inject()(countryOptions: CountryOptions) extends TransformedElement[I] {
```
Comment (grantrobson) [inline app/models/details/transformation/SchemeDetailsSection.scala]: This doesn't need to be a case class.

File: app/models/details/transformation/TransformedElement.scala [hunk 9]
```diff
@@ -16,17 +16,111 @@
...
-import viewmodels.AnswerRow
+import models.details._
+import utils.CountryOptions
+import viewmodels.{AnswerRow, SuperSection}
+
 import scala.language.implicitConversions
 
 trait TransformedElement[I] {
 
+  val entityType: String = "individual"
```
Comment (grantrobson) [inline app/models/details/transformation/TransformedElement.scala]: Make private/protected anything not needing to be public

File: app/models/details/transformation/TransformedElement.scala [hunk 10]
```diff
@@ -16,17 +16,111 @@
...
+      Some(s"${address.addressLine1},"),
+      Some(s"${address.addressLine2},"),
+      address.addressLine3.map(line3 => s"$line3,"),
+      address.addressLine4.map(line4 => s"$line4,"),
+      address.postalCode.map(postCode => s"$postCode,"),
+      Some(getCountry(countryOptions, address.countryCode))
+    ).flatten
+  }
+
+  protected def yesNo(flag: Boolean): String = {
```
Comment (grantrobson) [inline app/models/details/transformation/TransformedElement.scala]: Remove unnecessary brackets

File: app/models/details/transformation/TrusteeInfoSection.scala [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+package models.details.transformation
+
+import javax.inject.Inject
+import models.details._
+import viewmodels.MasterSection
+
+import scala.language.implicitConversions
+
+case class TrusteeInfoSection @Inject()(individualInfoRows :IndividualInfoRows[IndividualInfo],
```
Comment (grantrobson) [inline app/models/details/transformation/TrusteeInfoSection.scala]: Doesn't need to be case class

File: app/models/details/transformation/CompanyDetailsRows.scala (hunk 1 above)
Comment (abhijeetardale) [reply app/models/details/transformation/CompanyDetailsRows.scala]: fixed

File: app/models/details/transformation/IndividualInfoRows.scala (hunk 5 above)
Comment (abhijeetardale) [reply app/models/details/transformation/IndividualInfoRows.scala]: added for clean code and improve readability

File: app/models/details/transformation/TransformedElement.scala (hunk 9 above)
Comment (abhijeetardale) [reply app/models/details/transformation/TransformedElement.scala]: fixed

File: app/models/details/transformation/TransformedElement.scala (hunk 10 above)
Comment (abhijeetardale) [reply app/models/details/transformation/TransformedElement.scala]: fixed

File: app/models/details/transformation/TrusteeInfoSection.scala (hunk 11 above)
Comment (abhijeetardale) [reply app/models/details/transformation/TrusteeInfoSection.scala]: fixed

File: app/models/details/transformation/SchemeDetailsSection.scala (hunk 8 above)
Comment (abhijeetardale) [reply app/models/details/transformation/SchemeDetailsSection.scala]: fixed

File: app/models/details/transformation/IndividualInfoRows.scala (hunk 4 above)
Comment (abhijeetardale) [reply app/models/details/transformation/IndividualInfoRows.scala]: fixed

File: app/models/details/transformation/EstablisherInfoSection.scala (hunk 3 above)
Comment (abhijeetardale) [reply app/models/details/transformation/EstablisherInfoSection.scala]: fixed

File: app/models/details/transformation/CompanyDetailsRows.scala (hunk 2 above)
Comment (abhijeetardale) [reply app/models/details/transformation/CompanyDetailsRows.scala]: its already protected, coming from parent

**Pull request merged**